services/credentials.go

```diff
@@ -148,11 +149,11 @@ func (s *Service) CreateCredentialWithRetry(msg []byte, sig []byte, ot credentia
 
 // Creates a new credential for a node. If a valid credential exists, it will be returned instead.
 // No retry logic is implemented, so it is up to the caller to retry if it does not succeed.
-func (s *Service) CreateCredential(msg []byte, sig []byte, ot credentials.OperatorType) (*models.AuthenticatedCredential, error) {
+func (s *Service) CreateCredential(msg []byte, sig []byte, dataHash common.Hash, address common.Address, ot credentials.OperatorType) (*models.AuthenticatedCredential, error) {
 	var err error
 
 	// Validate request
-	nodeID, err := s.validateSignedRequest(&msg, &sig, ot)
+	nodeID, err := s.validateSignedRequest(&msg, &sig, &dataHash, &address, ot)
```

Comment:
And finally, passing those new datas into `validateSignedRequest`, which is where we'll determine how to verify the user-provided input data.